services/storage/src/simcore_service_storage/handlers.py

```diff
@@ -343,7 +347,11 @@ async def upload_file(request: web.Request):
                 source_uuid=source_uuid,
             )
         else:
-            link = await dsm.upload_link(user_id=user_id, file_uuid=file_uuid)
+            link = await dsm.upload_link(
```

Comment:
Obsservations:

1. if you look above there is this `extra_source` and `extra_location` which ware optional. You should raise an error if `link_type` is also present.
1. If you select dactore/pennsieve as a backend, the s3 links should probably be disabled both in upload and in download. Do you agree?

Reply:
yes. I will check that in the next PR as I anyway need to bring that s3 link also to the frontend.